src/scvi/dataloaders/_data_splitting.py

```diff
@@ -64,14 +61,15 @@ def validate_data_split(
             "any of the aforementioned parameters."
         )
 
-    if batch_size is not None and not drop_last:
-        if n_train % batch_size < 3 and n_train % batch_size > 0:
-            num_of_cells = n_train % batch_size
+    if batch_size_for_adaptive_last_batch is not None:
```

Comment:
Do we need this if drop_last is true?

Reply:
I'm not sure how drop_last is handled now?

Reply:
Warning should still be printed, if train size is not None.

Reply:
Best case you change the prodecure here. Keep batch_size and add another argument, adapt_train_size to highlight this use case.